libvast/src/system/source_command.cpp

```diff
@@ -99,11 +99,14 @@ caf::message source_command(const command& cmd, caf::actor_system& sys,
                ? caf::get<caf::actor>(node_opt)
                : caf::get<scope_linked_actor>(node_opt).get();
   VAST_DEBUG(&cmd, "got node");
-  /// Spawn an actor that takes care of CTRL+C and friends.
-  auto sig_mon = self->spawn<detached>(system::signal_monitor, 750ms,
-                                       actor{self});
+  // Start signal handler.
+  std::thread smon{[&] {
+    CAF_SET_LOGGER_SYS(&sys);
+    signal_monitor::run(750ms, self);
+  }};
   auto guard = caf::detail::make_scope_guard([&] {
-    self->send_exit(sig_mon, exit_reason::user_shutdown);
+    signal_monitor::stop = true;
+    smon.join();
```

Comment:
Just a thought: we're repeating this code pattern in `source_command` and `start_command`, maybe we can factor it into a static method in `signal_monitor` that would return the scoped guard?